mqtt/mqtt-broker/src/proptest.rs

```diff
@@ -0,0 +1,268 @@
+use std::time::Duration;
+
+use mqtt3::proto;
+use proptest::{
+    bool,
+    collection::{hash_map, hash_set, vec, vec_deque},
+    num,
+    prelude::*,
+};
+
+use crate::{
+    session::{IdentifiersInUse, PacketIdentifiers},
+    BrokerState, ClientId, Publish, Segment, SessionState, Subscription, TopicFilter,
+};
+use bytes::Bytes;
+
+prop_compose! {
+    pub fn arb_broker_state()(
+        retained in hash_map(arb_topic(), arb_publication(), 0..20),
+        sessions in vec(arb_session_state(), 0..10),
+    ) -> BrokerState {
+        BrokerState::new(retained, sessions)
+    }
+}
+
+prop_compose! {
```

Comment:
:+1:  nice to have all prop_compose in a separate module!

Reply:
Yeah, basically I borrowed approach from https://github.com/bodil/im-rs crate